### PR TITLE
[refactor](Nereids): refactor UT of Nereids

### PR DESCRIPTION
**Pull request by jackwener**
# Proposed changes

Issue Number: close #xxx

## Problem Summary:

refactor the UT of Nereids.

Extract the plan constructor (This PR extract all olapscan and table into `PlanConstructor `).

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [x] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [x] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [x] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [x] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes
    - [x] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: fe/fe-core/src/test/java/org/apache/doris/nereids/rules/implementation/LogicalProjectToPhysicalProjectTest.java [hunk 1]
```diff
@@ -18,28 +18,61 @@
...
+        return implPlan;
+    }
+
+
+    @Test
+    public void projectionImplTest(@Mocked PlannerContext plannerContext) {
+        LogicalOlapScan scan = PlanConstructor.newLogicalOlapScan("a");
+        LogicalPlan project = new LogicalProject<>(Lists.newArrayList(), scan);
+
+        Plan implPlan = rewriteLogicalToPhysical(project, plannerContext);
```
Comment (adonis0147) [inline fe/fe-core/src/test/java/org/apache/doris/nereids/rules/implementation/LogicalProjectToPhysicalProjectTest.java:75]:
```suggestion
        Plan plan = new Memo(project).getRoot().getLogicalExpression().getPlan();
        Plan implPlan = rewriteLogicalToPhysical((LogicalPlan) plan, plannerContext);

```

File: fe/fe-core/src/test/java/org/apache/doris/nereids/rules/implementation/LogicalProjectToPhysicalProjectTest.java [hunk 2]
```diff
@@ -18,28 +18,61 @@
...
+        String name = plan.getClass().getName();
+        Rule rule = rulesMap.get(name);
         List<Plan> transform = rule.transform(plan, plannerContext);
         Assertions.assertEquals(1, transform.size());
-        Plan implPlan = transform.get(0);
+        Assertions.assertTrue(transform.get(0) instanceof PhysicalPlan);
+        PhysicalPlan implPlan = (PhysicalPlan) transform.get(0);
+        List<Plan> children = Lists.newArrayList();
+        for (Plan child : plan.children()) {
+            children.add(rewriteLogicalToPhysical((LogicalPlan) child, plannerContext));
```
Comment (adonis0147) [inline fe/fe-core/src/test/java/org/apache/doris/nereids/rules/implementation/LogicalProjectToPhysicalProjectTest.java:63]:
```suggestion
            LogicalPlan subLogicalPlan = (LogicalPlan) ((GroupPlan) child).getGroup().getLogicalExpression().getPlan();
            children.add(rewriteLogicalToPhysical(subLogicalPlan, plannerContext));

```

**Review by adonis0147 — CHANGES_REQUESTED** [review]
Using this modification can fix the FE check.

**Review by 924060929 — APPROVED** [review]
LGTM

**Pull request merged**